website/static/js/comment.js

```diff
@@ -132,27 +147,121 @@ BaseComment.prototype.setupToolTips = function(elm) {
     });
 };
 
-BaseComment.prototype.fetch = function() {
+BaseComment.prototype.fetch = function(isCommentList, thread) {
     var self = this;
     var deferred = $.Deferred();
     if (self._loaded) {
         deferred.resolve(self.comments());
     }
+    if (thread !== undefined) {
+        return self.getThread(thread);
+    }
     $.getJSON(
         nodeApiUrl + 'comments/',
-        {target: self.id()},
+        {
+            page: self.page(),
+            target: self.id(),
+            rootId: self.rootId(),
+            isCommentList: (isCommentList || null)
+        },
         function(response) {
             self.comments(
-                ko.utils.arrayMap(response.comments.reverse(), function(comment) {
+                ko.utils.arrayMap(response.comments.reverse(), function (comment) {
                     return new CommentModel(comment, self, self.$root);
                 })
             );
+            if (isCommentList) {
+                self.discussionByFrequency(response.discussionByFrequency);
+                self.discussionByRecency(response.discussionByRecency);
+            }
             self.unreadComments(response.nUnread);
             deferred.resolve(self.comments());
+            self.configureCommentVisibility();
             self._loaded = true;
         }
     );
-    return deferred;
+    return deferred.promise();
+};
+
+BaseComment.prototype.getThread = function(thread_id) {
+    var self = this;
+    var deferred = $.Deferred();
+    if (self._loaded) {
+        deferred.resolve(self.comments());
+    }
+    var request = $.getJSON(nodeApiUrl + 'comment/' + thread_id + '/');
+    request.done(function(response){
+        self.comments([new CommentModel(response.comment, self, self.$root)]);
+        deferred.resolve(self.comments());
+        self.configureCommentVisibility();
+        self._loaded = true;
+    })
+    return deferred.promise();
+}
+
+BaseComment.prototype.configureCommentVisibility = function() {
+    var self = this;
+    var FILES = 'files';
+    var PANE = 'pane';
+    for (var c in self.comments()) {
+        var comment = self.comments()[c];
+        if (self.level > 0 && self.loading() === false) {
+            comment.isHidden(self.isHidden());
+            if (!self.isHidden() && self.page() === FILES) {
+                comment.title(self.title());
+            }
+            comment.loading(false);
+            continue;
+        }
+        if (comment.page() !== FILES || self.mode === PANE) {
+            comment.loading(false);
+            continue;
+        }
+        (function (cmt) {
```

Comment:
@sloria http://stackoverflow.com/questions/21819905/jquery-ajax-calls-in-for-loop